sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4189,6 +4189,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_AVRO_PREVENT_READING_INCORRECT_SAME_ENCODED_TYPES =
+    buildConf("spark.sql.legacy.avro.preventReadingIncorrectSameEncodedTypes")
```

Comment:
ideally turning on the legacy config should enable the legacy behavior. How about `spark.sql.legacy.avro.allowReadingWithIncompatibleSchema`?

Reply:
sounds good!